### PR TITLE
Add documentation for RNN transducer

### PR DESCRIPTION
**Pull request by vincentqb**
Add prototype RNN transducer loss to the documentation. This will need to be guarded off from releases.

Screen Shot 2021-01-11 at 11 48 48 AM

cc #1137

### CODE REVIEW

File: docs/source/index.rst
Comment (mthrok) [inline docs/source/index.rst:42]: I think this `prototype` is the correct module name.

File: docs/source/transducer.rst
Comment (mthrok) [inline docs/source/transducer.rst:3]: I think it's a good idea to note that prototype is nightly-only feature at the top of all the pages for prototype. So that there is a less chance of receiving the report like [this](https://github.com/pytorch/text/issues/1079)

File: docs/source/transducer.rst
Comment (mthrok) [inline docs/source/transducer.rst:7]: Did you verify the doc builds? Since prototype module has to be imported separately from `import torchaudio`, I think you need to add `import torchaudio.prototype` to doc config.

File: docs/source/index.rst
Comment (vincentqb) [reply docs/source/index.rst:42]: If you mean in terms of compilation, this does compile locally, see preview in description.

If you mean in terms of file organization, [torchtext](https://github.com/pytorch/text/tree/master/docs/source) also groups each prototype/experimental feature in its own doc file.

Is that what you meant?

File: docs/source/transducer.rst
Comment (vincentqb) [reply docs/source/transducer.rst:3]: I've labelled the issue as prototype and linked to the "source of truth" definition of prototype, which says "not part of release". Do you mean to say that you feel this may not be clear enough?

File: docs/source/transducer.rst
Comment (vincentqb) [reply docs/source/transducer.rst:7]: Yes, the compilation looks good to me. Do you see something missing from the preview in the description?

File: docs/source/transducer.rst
Comment (mthrok) [reply docs/source/transducer.rst:3]: Yes, I think we cannot be too careful here.

I suggest add `.. note::` or `.. warning::` and tell users to use nightly build in a more straightforward manner. 

Reading the classification, I feel like it is not even included in nightly build, which is another binary distribution.

> Prototype: These features are typically not available as part of binary distributions like PyPI or Conda, except sometimes behind run-time flags, and are at an early stage for feedback and testing.

File: docs/source/transducer.rst
Comment (mthrok) [reply docs/source/transducer.rst:7]: If it compiles, then that's okay. Let's see if the automated push also works.

File: docs/source/index.rst
Comment (mthrok) [reply docs/source/index.rst:42]: Looking at the `compliance.kaldi` I thought it had to be `prototype.transducer`. If the doc is compiled correctly then it's not an issue.

**Review by mthrok — APPROVED** [review]
Approving so that you are not blocked, can merge it and see how it is reflected at pytorch.org/audio

After the merge, you can run the nightly job manually from #306 and check how it looks on pytorch.org/audio. If you see an issue, feel free to open a new PR.

**Pull request merged**
